packages/DataTable/src/components/InfiniteLoader/InfiniteLoader.tsx

```diff
@@ -47,23 +50,49 @@ export function InfiniteLoaderImpl({
   threshold = 15,
 }: InfiniteLoaderPrivateProps & InfiniteLoaderPublicProps): JSX.Element {
   const infiniteLoaderRef = React.useRef(null);
-  const listRef = React.useRef(null);
+  const listRef = React.useRef<VariableSizeList>(null);
   const rowHeights = React.useRef<Record<number, number>>({});
+  const { allColumns } = useReactTableContext();
+  const columnsWidth = React.useMemo(
+    () =>
+      allColumns.reduce(
+        (res, column) => {
+          res[column.id] = column.totalWidth;
+          return res;
+        },
+        {} as TableColumnsWidth
+      ),
+    [allColumns]
+  );
 
   function getItemSize(index: number): number {
     if (!rowHeights.current[index]) {
-      const newRowHeight = calculateRowHeight(data[index]);
+      const newRowHeight = calculateRowHeight({
```

Comment:
I'm concerned that is may become a performance bottleneck the fact that we `append/remove` real DOM elements for every element.

Let's have a call and benchmark it together